### DISCUSSION TITLE
[Question] How to map indexable types?

### DISCUSSION THREAD

**Discussion by Edwinws** [nartc/mapper#445] [General]
How do we go around automapping [Indexable types](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures)?

```typescript
export class Blah {
    system_params!: { [key: string]: any };
}
```

```typescript
export class BlahDto {
    system_params!: { [key: string]: any };
}
```

I tried putting it into a class and using `@AutoMap({ typeFn: () => ParamClass })` but it doesn't work.

If it cannot work, what are the alternatives I can do if the keys are unknown? I can currently work around the problem by not automapping the field and perform a manual assignment after mapping.

**Comment by nartc** [comment]
@Edwinws Thanks for opening the issue. I can tell you from top of my head that it is due to TypeScript limitation. However, would you mind providing a somewhat real use-case to see if I can think of a way to support it? Personally, I haven't had the need to map Dictionary/Dynamic Object yet.

**Comment by Edwinws** [comment]
@nartc Thanks for the response! The field stores a key-value pair that is received from another system via API call, and forwarded by the current system to another system via API call (the current system does not actually utilize it). 

Unfortunately, I do not know the full list of the keys, and they can change at any time.

**Comment by nartc** [comment]
I see. I'll keep this one open and will think more about it. Since you have a workaround, I guess it's not too urgent. Thanks again for the issue. Feel free to chime in anything that you think it'd be helpful to get this supported.

**Comment by calvinwyoung** [comment]
@Edwinws Did you find a solution to this? Our team is running into the same issue right now

**Reply by Edwinws** [reply]
No I did not - not using the mapper. I simply did a manual assignment for the field.

**Comment by LaysDragon** [comment]
bump into this too,also automapper seems didn't support `Map` type?

**Comment by HarryPi** [comment]
Similar issue here sending a dictionary from the backend to the frontend, it seems its not capable of handling it

**Comment by maxime-aubry** [comment]
Hello @nartc 
Is there a solution using by you library ?
I am in the same case today.
